sootup.java.core/src/main/java/sootup/java/core/views/JavaView.java

```diff
@@ -87,13 +86,8 @@ public synchronized Stream<JavaSootClass> getClasses() {
 
     Stream<JavaSootClass> resolvedClasses =
         inputLocations.stream()
-            .flatMap(
-                location -> {
-                  // TODO: [ms] find a way to not stream().collect().stream()
-                  return location.getClassSources(this).toList().stream();
-                })
-            .map(sootClassSource -> (JavaSootClassSource) sootClassSource)
-            .map(this::buildClassFrom);
+            .flatMap(location -> location.getClassSources(this))
```

Comment:
does this still work with ondemand loading i.e. accessing the AnalysisInputLocation twice - as flatmap closes the underlying stream?